File: backend/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.pagination import PageParams, PaginatedResponse


class AuditLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    actor_user_id: int | None
    actor_user_name: str | None
    action: str
    entity_type: str
    entity_id: int | None
    ip_address: str | None
    user_agent: str | None
    metadata_json: dict | None
    created_at: datetime


class AuditLogListParams(PageParams):
    actor_user_id: int | None = None
    action: str | None = None
    entity_type: str | None = None
    entity_id: int | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None
    search: str | None = None


class AuditLogListResponse(PaginatedResponse[AuditLogResponse]):
    pass

File: frontend/src/types/audit.ts
export interface AuditLog {
  id: number;
  actor_user_id: number | null;
  actor_user_name: string | null;
  action: string;
  entity_type: string;
  entity_id: number | null;
  ip_address: string | null;
  user_agent: string | null;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
}

export interface AuditLogFilters {
  page?: number;
  page_size?: number;
  actor_user_id?: number;
  action?: string;
  entity_type?: string;
  entity_id?: number;
  date_from?: string;
  date_to?: string;
  search?: string;
}

export interface AuditLogListResponse {
  items: AuditLog[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}
